src/python/zquantum/core/circuit/_gate_test.py

```diff
@@ -1,7 +1,184 @@
 import unittest
 import os
 from ._gate import Gate
+from ._gateset import COMMON_GATES, UNIQUE_GATES
+from ._qubit import Qubit
+from sympy import Symbol
+import qiskit
+from math import pi
+
 
 class TestGate(unittest.TestCase):
-    def test_test(self):
-        pass
\ No newline at end of file
+    def setUp(self):
+        self.two_qubit_gates = ["CNOT", "CZ", "CPHASE", "SWAP"]
+        self.one_parameter_gates = ["PHASE", "Rx", "Ry", "Rz", "CPHASE"]
+
+    def create_gate(self, gate_name, params=None):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        if params is None:
+            params = []
```

Comment:
Because there is this issue when you pass empty lists or dictionaries as input params in Python. I don't remember why exactly it causes problems, but it's better not to do that.